File: FoodOrderApis/OrderService/Data/Models/Food.cs
namespace OrderService.Data.Models;

public class Food
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Describe { get; set; }
    public string ImageUrl { get; set; }
    public List<OrderDetail> OrderDetails { get; set; }
}

File: food_order_app/lib/data/models/food.dart
class Food {
  int id;
  String name;
  String imageUrl;
  String describe;
  double? price;
  String? userId;

  Food(
      {required this.id,
      required this.name,
      required this.imageUrl,
      required this.describe,
      required this.price,
      required this.userId});

  factory Food.fromJson(Map<String, dynamic> json) {
    return Food(
        id: json['id'],
        name: json['name'],
        imageUrl: json['imageUrl'],
        describe: json['describe'],
        price: json['price'] != null ? json['price'] / 1.0 : null,
        userId: json['userId'] ?? json['userId']);
  }
}
